### ISSUE TITLE
Semgrep Rule for Nginx missing-internal.yaml causes Segmentation Fault

### ISSUE THREAD

**Issue by bp-gilley-justin** [returntocorp/sgrep-rules#3726]
**Describe the bug**
When using [missing-internal.yaml](https://github.com/semgrep/semgrep-rules/blob/develop/generic/nginx/security/missing-internal.yaml) rule, a segmentation fault can occur with an *nginx.conf* file: see attached [nginx.txt](https://github.com/user-attachments/files/24133398/nginx.txt).
Github does not allow *.conf* extension, so I renamed extension to *.txt*.

**To Reproduce**
Using *Python 3.12.3* and *Ubuntu 24.04.3 LTS* (on WSL).  This was reproduced on an *Alpine 3.22* container with *Python 3.12.12* as well.
Used *BASH 5.2.21* shell
```bash
python3 -m venv .venv
. .venv/bin/activate
pip install semgrep==1.145.1

wget https://github.com/semgrep/semgrep-rules/raw/refs/heads/develop/generic/nginx/security/missing-internal.yaml
# get attached nginx.txt and rename it to nginx.conf
.venv/bin/semgrep scan --config missing-internal.yaml nginx.conf
```

**Expected behavior**
Running [nginx.txt](https://github.com/user-attachments/files/24133398/nginx.txt) (with name *nginx.conf*) should not produce a segmentation fault and display rule violations (if applicable).

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Additional Context**
Line 40 on the nginx configuration file is blank, and if I replace it with *internal;*, then the file is processed as expected by *semgrep*.  I do not know much about *semgrep* internals, but I presume *pattern-not-inside* match prevents the code block that causes the segmentation fault to *not* run.
